src/puppetlabs/puppetdb/scf/storage.clj

```diff
@@ -1114,28 +1115,21 @@
   "Prep `event` for comparison/computation of a hash"
   [event]
   (-> event
-      (utils/update-when [:timestamp] to-timestamp)
-      (utils/update-when [:old_value] sutils/db-serialize)
-      (utils/update-when [:new_value] sutils/db-serialize)
+      (update :timestamp to-timestamp)
+      (update :old_value sutils/db-serialize)
+      (update :new_value sutils/db-serialize)
```

Comment:
Yes every invocation definitely goes through this check or has these keys in the map, e.g. our test reports.